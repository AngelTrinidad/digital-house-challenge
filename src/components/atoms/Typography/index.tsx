import React, {FC} from 'react';
import {TextProps, TextStyle} from 'react-native';

import {TypographyVariant, Variant} from './types';

import Text from '~components/atoms/Text';
import {useTheme} from '~config/themes';

export type {TypographyVariant};

interface Props extends TextProps {
  variant?: TypographyVariant;
  color?: 'primary' | 'secondary';
}

const VARIANT_STYLES: {[key: string]: TextStyle} = {
  [Variant.xxl]: {
    fontSize: 24,
  },
  [Variant.xl]: {
    fontSize: 20,
  },
  [Variant.lg]: {
    fontSize: 16,
  },
  [Variant.normal]: {
    fontSize: 14,
  },
  [Variant.sm]: {
    fontSize: 12,
  },
};

const Typography: FC<Props> = ({
  children,
  style,
  variant = 'normal',
  color = 'primary',
  ...props
}) => {
  const {
    colors: {text},
  } = useTheme();

  const textStyle: TextStyle = {
    ...VARIANT_STYLES[variant],
    color: text[color],
    fontWeight: '800',
  };

  return (
    <Text style={[textStyle, style]} {...props}>
      {children}
    </Text>
  );
};

export default Typography;
