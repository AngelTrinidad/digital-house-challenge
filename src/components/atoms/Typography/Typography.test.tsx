import React from 'react';

import {render} from '@testing-library/react-native';

import Typography from '.';

describe('<Typography />', () => {
  let component: any;

  beforeEach(() => {
    component = render(<Typography>Hello</Typography>);
  });

  it('should renders correctly', () => {
    expect(component).toBeDefined();
  });

  it('should has the same snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should convert variant to correct font size', () => {
    const XXLComponentJSON = render(
      <Typography variant="xxl" />,
    ).toJSON() as any;
    const XLComponentJSON = render(<Typography variant="xl" />).toJSON() as any;
    const LGComponentJSON = render(<Typography variant="lg" />).toJSON() as any;
    const SMComponentJSON = render(<Typography variant="sm" />).toJSON() as any;

    expect(XXLComponentJSON.props.style).toMatchObject({
      fontSize: 24,
    });

    expect(XLComponentJSON.props.style).toMatchObject({
      fontSize: 20,
    });

    expect(LGComponentJSON.props.style).toMatchObject({
      fontSize: 16,
    });

    expect(SMComponentJSON.props.style).toMatchObject({
      fontSize: 12,
    });
  });
});
