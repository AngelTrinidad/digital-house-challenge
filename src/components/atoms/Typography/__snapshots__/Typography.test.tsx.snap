// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typography /> should has the same snapshot 1`] = `
<Text
  style={
    {
      "color": "black",
      "fontFamily": "Metropolis Bold",
      "fontSize": 14,
      "fontWeight": "800",
    }
  }
>
  Hello
</Text>
`;
